### DISCUSSION TITLE
Is it possible for multiple different widgets to subscribe to the same stream?

### DISCUSSION THREAD

**Discussion by justinmoon** [fzyzcjy/flutter_rust_bridge#790] [General]
My app has an event loop the receives incoming notifications from a server.

I think I understand how to modify the [logging example](https://cjycode.com/flutter_rust_bridge/feature/logging.html) to have one widget subscribe to events.

Would it be possible to allow multiple different widgets to subscribe to these events separately? Or would I need one thing to make the subscription and distribute events on the dart side?

**Comment by fzyzcjy** [comment]
Sure. Just write down a few lines in Dart to wrap the native stream. Btw Stream.broadcast may be helpful
